adafruit_imageload/png.py

```diff
@@ -107,22 +108,37 @@ def load(  # noqa: PLR0912, PLR0915, Too many branches, Too many statements
     if mode == 3:  # indexed
         bmp = bitmap(width, height, colors)
         mem = memoryview(bmp)
+        pixels_per_byte = 8 // depth
+        # Adjust for Displayio.Bitmap filler to scanline at 4-byte boundry
+        filladj = (4 - (scanline % 4)) % 4
+        dst = 0
+        src = 1
+        src_b = 1
+        if (
+            (implementation[1][0] == 9 and implementation[1][1] < 2) or implementation[1][0] < 9
+        ) and (depth < 8 or width % 4 != 0):
+            workaround = True
+        else:
+            workaround = False
+
         for y in range(height):
-            dst = y * scanline
-            src = y * (scanline + 1) + 1
-            if depth < 8:
+            if workaround:
```

Comment:
you probably don't want to check this condition on every iteration of the loop, better move it outside the loop and duplicate the loop line